server/plugin.go

```diff
@@ -635,22 +706,87 @@ func (p *Plugin) HasGroupHook(ctx context.Context, user *gitlab.UserInfo, namesp
 	return found, err
 }
 
-func (p *Plugin) checkAndRefreshToken(token *oauth2.Token) (*oauth2.Token, error) {
-	// If there is only one minute left for the token to expire, we are refreshing the token.
-	// The detailed reason for this can be found here: https://github.com/golang/oauth2/issues/84#issuecomment-831492464
-	// We don't want the token to expire between the time when we decide that the old token is valid
-	// and the time at which we create the request. We are handling that by not letting the token expire.
-	if time.Until(token.Expiry) <= 1*time.Minute {
-		conf := p.getOAuthConfig()
-		src := conf.TokenSource(context.Background(), token)
-		newToken, err := src.Token() // this actually goes and renews the tokens
-		if err != nil {
-			return nil, errors.Wrap(err, "unable to get the new refreshed token")
+func (p *Plugin) refreshToken(userInfo *gitlab.UserInfo, token *oauth2.Token) (*oauth2.Token, error) {
+	conf := p.getOAuthConfig()
+	src := conf.TokenSource(context.Background(), token)
+
+	newToken, err := src.Token() // this actually goes and renews the tokens
+
+	if err != nil {
+		if strings.Contains(err.Error(), "\"error\":\"invalid_grant\"") {
```

Comment:
Nit: Should the error message be a global const?

Reply:
I didn't because it's only used in one place, but it's a matter of your preferred conventions.

Reply:
> I didn't because it's only used in one place, but it's a matter of your preferred conventions.

When a constant has a significant meaning like this (essentially an error id in the external system), we generally put it at the top of the file to show that it is truly a constant